sdk/messaging/eventgrid/aznamespaces/shared_test.go

```diff
@@ -181,72 +179,75 @@ func newReceiverClient(t *testing.T, useSASKey bool) *aznamespaces.ReceiverClien
 }
 
 func newRecordingTransporter(t *testing.T, testVars testVars) policy.Transporter {
-	transport, err := recording.NewRecordingHTTPClient(t, nil)
-	require.NoError(t, err)
+	recordingOpts := &recording.RecordingOptions{
+		ProxyPort:    os.Getpid()%10000 + 20000,
```

Comment:
This makes me nervous because it depends on the implementation of `recording.StartTestProxy`. The only non-default value in your `RecordingOptions` is `TestInstance`, and you only need to set that if you need to apply sanitizers to a single `testing.T`. It appears you don't need to do that i.e., you apply the same sanitizers to all recordings anyway, so I think you could just use the default `RecordingOptions`.

Reply:
I'm definitely okay with changing it. I was having issues with my sanitizers even firing off and this "fixed" it.

I wonder if, instead of setting up sanitizers over and over I can just do it once for the entire run instead - is that typically what people do? Or is it supposed to be safe to apply them over and over again?

Reply:
Adding the same sanitizers to every test individually isn't incorrect, it's just inefficient. If the sanitizers don't break a test, you can add them for all tests (zero `TestInstance`) immediately after starting the proxy. Every other test suite does this today because we don't have a sanitizer that's required for one test but that breaks another.

Reply:
I think the primary problem was because I was adding in sanitizers after I called recording.Start(). I fixed that and now the options aren't needed anymore.

Also, I changed it so the sanitizers only initialize once. I'm not yet intentionally trying to create per-test sanitizers.